lib/src/cli.dart

```diff
@@ -74,28 +75,40 @@ ArgParser createArgsParser() {
       help: "print the version of grinder");
   parser.addFlag('help', abbr: 'h', negatable: false,
       help: "show targets but don't build");
-  parser.addFlag('deps', abbr: 'd', negatable: false,
+  // TODO: Deprecated; remove post 0.7.0.
+  parser.addFlag('deps', abbr: 'd', negatable: false, hide: true,
       help: "display the dependencies of targets");
   return parser;
 }
 
-void printUsage(ArgParser parser, Grinder grinder) {
+void printUsageAndDeps(ArgParser parser, Grinder grinder) {
   print('usage: dart ${currentScript()} <options> target1 target2 ...');
   print('');
   print('valid options:');
   print(parser.usage.replaceAll('\n\n', '\n'));
 
-  if (!grinder.tasks.isEmpty) {
+  if (grinder.tasks.isEmpty) {
+    print('');
+    print('no current grinder targets');
+  } else {
+    // calculate the dependencies
+    grinder.start([], dontRun: true);
+
+    print('');
+    print('targets:');
     print('');
-    print('valid targets:');
 
     List<GrinderTask> tasks = grinder.tasks.toList();
-    tasks.forEach((t) {
-      var buffer = new StringBuffer()..write('  $t');
-      if (grinder.defaultTask == t) buffer.write(' (default)');
-      if (t.description != null) buffer.write(' ${t.description}');
-      print(buffer.toString());
-    });
+    print(tasks.map((task) {
+      bool isDefault = grinder.defaultTask == task;
+      Iterable<GrinderTask> deps = grinder.getImmediateDependencies(task);
+
+      String str = '${task}\n';
+      if (isDefault) str += '  @DefaultTask\n';
```

Comment:
I see how referencing the annotations in command-line help would be nice for devs familiar with the annotations.  But a dev who just wants to run the tasks, and doesn't know or want to know anything about how they are implemented, shouldn't have to, and might find it more difficult to decipher their meaning than if we used a more natural language approach.  We still have the programmatic `addTask` and `defaultTask = ...`, so the task could have been added that way instead.

So my preference would be something like:
```
[test]                  Run the tests.
[analyze]               Analyze the source code.
[buildbot] (default)    Tasks to run on the build bot.             
 ┣ [analyze]
 ┣ [test]
 ┣ [check-init]
 ┗ [coverage]
```

I like how you didn't show transitive dependencies, since I think that could get too busy / redundant.